snap.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019 Canonical Ltd.
+// Copyright 2015 Cloudbase Solutions SRL
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package manager
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/proxy"
+)
+
+// snap is the PackageManager implementation for snap-based systems.
+type snap struct {
+	basePackageManager
+}
+
+var snapMissingConfig = regexp.MustCompile(`error: snap "[a-z-]+" has no "[a-z-]+" configuration option`)
+
+// Search is defined on the PackageManager interface.
+func (snap *snap) Search(pack string) (bool, error) {
+	out, _, err := RunCommandWithRetry(snap.cmder.SearchCmd(pack), nil)
+	if err != nil {
+		return false, err
+	}
+
+	if strings.HasPrefix(out, "No matching snaps") {
+		return false, nil
+	}
+	return true, nil
+}
+
+// Install is defined on the PackageManager interface.
+func (snap *snap) Install(packs ...string) error {
+	fatalErr := func(output string) error {
+		if strings.Contains(output, "not found") {
+			return errors.New("unable to locate package") // error message from apt
+		}
+		return nil
+	}
+	_, _, err := RunCommandWithRetry(snap.cmder.InstallCmd(packs...), fatalErr)
+	return err
+}
+
+// GetProxySettings is defined on the PackageManager interface.
+func (snap *snap) GetProxySettings() (proxy.Settings, error) {
+	var res proxy.Settings
+
+	err := snap.readProxySettingsFromSnapSystemConfig(&res)
+	if err != nil {
+		return proxy.Settings{}, err
+	}
+	if res != (proxy.Settings{}) {
+		return res, nil
+	}
+
+	err = snap.readEtcEnvironmentForProxySettings(&res)
```

Comment:
As above, we don't care about the system proxy settings, just those set for snapd.